### PR TITLE
PWX-37684 | Add missing field for Volume configuration in helm

### PR DESCRIPTION
**Pull request by hitesh-wani-px**
**What this PR does / why we need it**:
Add missing field for Volume configuration in helm storageCluster template
**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PWX-37684

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by hitesh-wani-px** [comment]
**Testing Notes:**
1. Add following values in `values.yaml`
```yaml
volumes:                            
  - name: test-volume
    mountPath: /etc/test
    mountPropagation: Bidirectional  
    hostPath:
      path: /test
internalKVDB: true
```
2. Install PX
```
helm install my-release --set internalKVDB=true,clusterName=$(uuidgen) ./charts/portworx/ -n portworx --debug
```
3. The generated storageCluster template should incorporate all specified values.
```yaml
kind: StorageCluster
apiVersion: core.libopenstorage.org/v1
metadata:
  name: "mycluster"
  namespace: kube-system
  annotations:
  labels:
    heritage: "Helm"
    release: "my-release"
    chart: "portworx-3.1.0"
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/instance: "my-release"
spec:
  image: portworx/oci-monitor:3.1.0
  imagePullPolicy: Always

  kvdb:
    internal: true
  storage:
    useAll: true
  secretsProvider: k8s

  env:
  - name: PX_SECRETS_NAMESPACE
    value: portworx

  stork:
    enabled: true
  volumes:
  - hostPath:
      path: /test
    mountPath: /etc/test
    mountPropagation: Bidirectional
    name: test-volume
  csi:
    enabled: false
```
3. We should be able to see the following configurations reflected on the cluster:
    1. we should able to see `user-test-volume` in volume section of px pod
    ```yaml
    volumes:
    - hostPath:
         path: /test
         type: ""
      name: user-test-volume
    ```
    2. volumeMount path should be added for `test-volume` in px pod
    ```
    volumeMounts:
    - mountPath: /etc/test
      mountPropagation: Bidirectional
      name: user-test-volume
    ```
**Unit test cases results:**
```
--- PASS: TestStorageClusterHelmTemplate (0.00s)
    --- PASS: TestStorageClusterHelmTemplate/TestVolumes (0.07s)
    --- PASS: TestStorageClusterHelmTemplate/TestAllComponentsEnabled (0.07s)
    --- PASS: TestStorageClusterHelmTemplate/TestCustomRegistry (0.07s)
    --- PASS: TestStorageClusterHelmTemplate/TestExternalETCD (0.07s)
    --- PASS: TestStorageClusterHelmTemplate/TestPlacementTolerations (0.07s)
    --- PASS: TestStorageClusterHelmTemplate/TestDefaultChartValues (0.08s)
PASS
```

File: charts/portworx/values.yaml [hunk 1]
```diff
@@ -89,6 +89,7 @@ kvdb:
 volumes:                              # Add volumes to Portworx container. Supported volume types: Host, Secret, ConfigMap
   # - name: objectstore-cert
   #   mountPath: /etc/pwx/objectstore-cert
+  #   mountPropagation: None|HostToContainer|Bidirectional  
   #   secret:
```
Comment (nikita-bhatia) [inline charts/portworx/values.yaml:93]: @hitesh-wani-px are we missing readOnly field here?

Comment (hitesh-wani-px) [reply charts/portworx/values.yaml:93]: Done added now.

**Review by nikita-bhatia — APPROVED** [review]
LGTM

**Pull request merged**